PT-1860 make pt-osc respect case insesitive lookup on Windows and osx

ikruglov opened the pull request:
This PR teaches pt-osc how to perform case-insensitive lookup on platform which use case insensitive filesystems, such Windows or osx. Before searching the table via `SHOW TABLES FROM`, pt-osc checks `lower_case_table_names` param and  adjust it string comparison accordingly if `lower_case_table_names > 0`.

According to https://dev.mysql.com/doc/refman/8.0/en/identifier-case-sensitivity.html MySQL 5.6/5.7/8.0 are use case-sensitive search when `lower_case_table_names == 0` and case-insensitive lookup when value is either `1` or `2`.

Before the patch, pt-osc used to fail with `Table does not exist error`.

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=516) <br/>All committers have signed the CLA.

svetasmirnova reviewed (approved):
Please sign CLA, so we can accept your patch.

ikruglov commented:
Hi @svetasmirnova. I signed CLA. Please check it out. Hope that we can merge this PR.

percona-csalguero commented:
Hi @ikruglov .
The PR was made with the user `ikruglov-db`. You need to sing the CLA with that user. 
Thanks

svetasmirnova commented on bin/pt-online-schema-change:
```diff
@@ -3468,6 +3468,22 @@ sub check_table {
    }
    my ($dbh, $db, $tbl) = @args{@required_args};
    my $q      = $self->{Quoter} || 'Quoter';
+
+   # https://dev.mysql.com/doc/refman/8.0/en/identifier-case-sensitivity.html
+   # MySQL may use use case-insensitive table lookup, this is controller by
+   # @@lower_case_table_names. 0 means case sensitive search, 1 or 2 means
+   # case insensitive lookup.
+
+   my $lctn_sql = 'SELECT @@lower_case_table_names';
+   PTDEBUG && _d($lctn_sql);
+   my ($lower_case_table_names) = $dbh->selectrow_array($lctn_sql);
+   PTDEBUG && _d("lower_case_table_names=$lower_case_table_names");
+
+   if ($lower_case_table_names > 0) {
+       PTDEBUG && _d("MySQL uses case-insensitive lookup, converting '$tbl' to lowercase");
+       $tbl = lc $tbl;
+   }
+
```
This code is in the TableParser package and should be defined in lib/TableParser.pm Otherwise, it will be overwritten once we update the modules. Please adjust your PR.

ikruglov replied:
@svetasmirnova, I've updated the TableParse. Please check it out

ikruglov committed:
PT-1860 make pt-osc respect case insesitive lookup on Windows and osx

svetasmirnova committed:
Merge branch '3.x' into HEAD

svetasmirnova committed:
PT-1860 make pt-osc respect case insesitive lookup on Windows and osx #516

- Run update-modules
- Adjusted fix, so it does not create extra variable to hold $EVAL_ERROR
- Adjusted test case, so it runs on Linux
